### DISCUSSION TITLE
Use Lightbox component with an InfiniteScroll PhotoAlbum

### DISCUSSION THREAD

**Discussion by nimaen** [igordanchenko/react-photo-album#178] [Q&A] [answered]
Hi,

I wonder if it is possible to use a Lightbox component with an PhotoAlbum wrapped in an InfiniteScroll?
As much as I could see, the APIs diverge when we start to define a fetch function for the InfiniteScroll component.

Best,

Arnaud

**Comment by igordanchenko** [comment, marked as answer]
Hi there!

Yes, it's possible to use the `Lightbox` component with the `InfiniteScroll`. You'd just need to keep a copy of all the fetched photos.

https://stackblitz.com/edit/react-photo-album-178?file=src%2Fcomponents%2FRowsExample.tsx

I'll think about simplifying this integration. I'll probably need to add an `onClick` callback to the `InfiniteScroll` component that also provides a current copy of the `photos` array:


```jsx
<InfiniteScroll
  // ...
  onClick={({ photos, photo, index }) => ...}
>
```

**Reply by igordanchenko** [reply]
The `onClick` callback is now available in 3.0.1

Example - https://stackblitz.com/edit/react-photo-album-178-1?file=src%2Fcomponents%2FRowsExample.tsx

**Comment by nimaen** [comment]
Thank you for the swift reply!

I wondered if I could bypass this solution in fact by having a sort of common storage area for both the Lightbox and InfiniteScroll/PhotoAlbum.
But I'll do it this way for now, that would be ok, I guess.
I've tried 10k images in an infinite scroll, it worked pretty well until the browser started to slow, somehow.

I'm also developing another paginated view to compare the UX & performances.

**Reply by igordanchenko** [reply]
Please feel free to share a live demo if possible. I'd be curious to test the performance on a real-life example.

**Reply by flynnpc-swca** [reply]
Hello. I also have a requirement to use infinite scroll on a very large data set. We are trying to only keep 100-300 images in the Photo array at a time, fetching as needed. Your component is great! However, I was wondering if there was a way to execute fetch when scrolling towards the top of the array? Possibly with the original index? That way we could create a map to keep track of which photos to fetch again. Thank you for hearing me out.

**Reply by igordanchenko** [reply]
@flynnpc-swca it may be worthwhile to open a separate discussion with a specific example, as I'm unclear about your implementation approach. The `InfiniteScroll` component keeps all fetched photos internally (not the images, just the `Photo` objects), so I'm not sure how you are trying to keep only 100-300 images at a time.
